backend/extensions/Squidex.Extensions/Actions/SignalR/SignalRAction.cs

```diff
@@ -0,0 +1,92 @@
+        [Display(Name = "Action", Description = "BROADCAST -> send to all User, USER => send to one user, USERS => send to list of users, GROUP => send to group, GROUPS => send to list of groups.")]
+        [DataType(DataType.Text)]
+        public ActionTypeEnum ActionType { get; set; }
+
+        [Display(Name = "Methode Name", Description = "Set the Name of the hub method received by the customer, default value 'push.")]
+        [DataType(DataType.Text)]
+        public string MethodName { get; set; }
+
+        [Display(Name = "User", Description = "Set for notity one user by Id (command 'user'), one id by line for notity multi user (command 'users').")]
+        [DataType(DataType.MultilineText)]
+        [Formattable]
+        public string User { get; set; }
+
+        [Display(Name = "Group", Description = "Set for notity one group by Name (Command 'group'), one id by line for notity multi groups (command 'groups').")]
+        [DataType(DataType.MultilineText)]
+        [Formattable]
+        public string Group { get; set; }
+
+        [Display(Name = "Payload (Optional)", Description = "Leave it empty to use the full event as body.")]
+        [DataType(DataType.MultilineText)]
+        [Formattable]
+        public string Payload { get; set; }
+
+        protected override IEnumerable<ValidationError> CustomValidate()
+        {
+            if (!string.IsNullOrWhiteSpace(HubName) && !Regex.IsMatch(HubName, "^[a-z][a-z0-9]{2,}(\\-[a-z0-9]+)*$"))
+            {
+                yield return new ValidationError("Hub must be valid azure hub name.", nameof(HubName));
+            }
+
+            if (ActionType == ActionTypeEnum.USER && (string.IsNullOrWhiteSpace(User) || User.IndexOf("\n", System.StringComparison.OrdinalIgnoreCase) >= 0))
+            {
+                yield return new ValidationError("Group must be specified and unique with 'USER' Action.", nameof(HubName));
+            }
+
+            if (ActionType == ActionTypeEnum.USERS && (string.IsNullOrWhiteSpace(User) || User.IndexOf("\n", System.StringComparison.OrdinalIgnoreCase) < 0))
+            {
+                yield return new ValidationError("User must be specified and multiple with 'USERS' Action.", nameof(HubName));
+            }
+
+            if (ActionType == ActionTypeEnum.GROUP && (string.IsNullOrWhiteSpace(Group) || Group.IndexOf("\n", System.StringComparison.OrdinalIgnoreCase) >= 0))
+            {
+                yield return new ValidationError("Group must be specified and unique with 'GROUP' Action.", nameof(HubName));
+            }
+
+            if (ActionType == ActionTypeEnum.GROUPS && (string.IsNullOrWhiteSpace(Group) || Group.IndexOf("\n", System.StringComparison.OrdinalIgnoreCase) < 0))
+            {
+                yield return new ValidationError("Group must be specified and multiple with 'GROUPS' Action.", nameof(HubName));
+            }
+        }
+    }
+
+    public enum ActionTypeEnum
+    {
+        BROADCAST,
+        USER,
```

Comment:
Can you not make User/Users and Group/Groups dynamic? Depending how many users have been defined?

Reply:
I change the code and action for that